internal/reconcilers/pipeline_dependencies.go

```diff
@@ -163,3 +205,52 @@ func (r *PipelineDependenciesReconciler) deleteManifests(ctx context.Context, fi
 
 	return nil
 }
+
+func (r *PipelineDependenciesReconciler) modifyAndSaveTempClusterRoleBinding(_ context.Context, fileName string, owner, obj client.Object) (string, error) {
+	log := r.Log.WithName("modifyAndSaveTempClusterRoleBinding")
+
+	b, err := os.ReadFile(fileName)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Couldn't read manifest file for: %s", fileName))
+		return "", err
+	}
+
+	if err = yamlutil.Unmarshal(b, &obj); err != nil {
+		log.Error(err, fmt.Sprintf("Couldn't unmarshall yaml file for: %s", fileName))
+		return "", err
+	}
+
+	// type asserting to ensure proper kind inorder to update specific values before updating/creating in cluster
+	crb, ok := obj.(*rbacv1.ClusterRoleBinding)
+	if !ok {
+		return "", fmt.Errorf("could not type assert client.Object as ClusterRoleBinding")
```

Comment:
Is this function going to be called on non-ClusterRoleBindings? If so, do we want to handle this error differently? In practice, this shouldn't be run on things that aren't the ClusterRoleBinding, so an error here should be a no-op for other object types, no?

Reply:
This method only called for `ClusterRoleBindings` since this is the only file that is templated, ie we need to replace the CRB's `name` and reference `namespace` for the `ServiceAccount` so everything is linked properly. If I don't do a type assertion I can't modify the object how I need to.

Reply:
The type assertion is fine. I'm just unsure if we need the error here, or we should treat it as a no-op and do nothing (i.e. don't emit an error). I suppose if we're handling the error such that "wrong type" is just ignored at call time, then this question doesn't matter.